manifests/install.pp

```diff
@@ -1,18 +1,23 @@
+# Class networkmanager::install
 class networkmanager::install {
-  package { 'network-manager':
-    ensure => $::networkmanager::version,
-  }
-  case $::networkmanager::gui {
-    'gnome': {
-      package { 'network-manager-gnome':
-        ensure => $::networkmanager::version,
-      }
+  include ::networkmanager
+
+  if $::networkmanager::manage_packages {
+    package { $::networkmanager::package:
+      ensure => $::networkmanager::package_ensure,
     }
-    'kde': {
-      package { 'plasma-nm':
-        ensure => present,
+    case $::networkmanager::gui {
+      'gnome': {
+        package { $::networkmanager::package_gnome:
```

Comment:
If that's a clear branch with exclusive branches, there's no point in having two `package` resources. One resource with a fixed title and a namevar should do.